test/common/upstream/load_balancer_impl_test.cc

```diff
@@ -540,6 +568,99 @@ TEST_P(LoadBalancerBaseTest, BoundaryConditions) {
   }
 }
 
+class TestZoneAwareLb : public ZoneAwareLoadBalancerBase {
+public:
+  TestZoneAwareLb(const PrioritySet& priority_set, ClusterStats& stats, Runtime::Loader& runtime,
+                  Random::RandomGenerator& random,
+                  const envoy::config::cluster::v3::Cluster::CommonLbConfig& common_config)
+      : ZoneAwareLoadBalancerBase(priority_set, nullptr, stats, runtime, random, common_config) {}
+
+  // ZoneAwareLoadBalancerBase will keep a copy of cross priority host map shared pointer and update
+  // it when the membership is updated.
+  const HostMapConstSharedPtr& crossPriorityHostMapForTest() { return cross_priority_host_map_; }
+
+  HostConstSharedPtr chooseHostOnce(LoadBalancerContext*) override {
+    return choose_host_once_host_;
+  }
+  HostConstSharedPtr peekAnotherHost(LoadBalancerContext*) override {
+    NOT_IMPLEMENTED_GCOVR_EXCL_LINE;
+  }
+
+  HostConstSharedPtr choose_host_once_host_{std::make_shared<NiceMock<MockHost>>()};
+};
+
+// Used to test common functions of ZoneAwareLoadBalancerBase.
+class ZoneAwareLoadBalancerBaseTest : public LoadBalancerTestBase {
+public:
+  envoy::config::cluster::v3::Cluster::CommonLbConfig common_config_;
+  TestZoneAwareLb lb_{priority_set_, stats_, runtime_, random_, common_config_};
+};
+
+TEST_F(ZoneAwareLoadBalancerBaseTest, CrossPriorityHostMapUpdate) {
+  // Fake cross priority host map.
+  auto host_map = std::make_shared<HostMap>();
+  priority_set_.cross_priority_host_map_ = host_map;
+
+  // Mock membership update and priority update callbacks will be executed.
+  host_set_.runCallbacks({}, {});
+
+  // Host map in the lb is updated.
+  EXPECT_EQ(host_map.get(), lb_.crossPriorityHostMapForTest().get());
+}
+
+TEST_F(ZoneAwareLoadBalancerBaseTest, SelectOverrideHostTestInLb) {
+  NiceMock<Upstream::MockLoadBalancerContext> context;
+
+  {
+    LoadBalancerContext::OverrideHost override_host{"1.2.3.4", HealthyStatus | DegradedStatus};
+    EXPECT_CALL(context, overrideHostToSelect())
+        .WillOnce(Return(absl::make_optional(override_host)));
+
+    // Mock membership update and update host map shared pointer in the lb.
+    auto host_map = std::make_shared<HostMap>();
+    priority_set_.cross_priority_host_map_ = host_map;
+    host_set_.runCallbacks({}, {});
+
+    // Expected host is not exist in the host map and then `chooseHostOnce` will be called.
```

Comment:
I'd do this as `// The expected host does not exist in the host map, therefore ...`

Reply:
Thanks.